test/GPv2Settlement/ComputeTradeExecutions.FilledAmounts.t.sol

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+pragma solidity ^0.8.26;
+
+import {BaseComputeTradeExecutions, GPv2Order, SettlementEncoder} from "./ComputeTradeExecutions.t.sol";
+import {Order} from "test/libraries/Order.sol";
+
+contract OrderFilledAmounts is BaseComputeTradeExecutions {
+    using GPv2Order for GPv2Order.Data;
+    using Order for GPv2Order.Data;
+    using SettlementEncoder for SettlementEncoder.State;
+
+    function test_should_fill_full_sell_amount_for_fill_or_kill_sell_orders() public {
+        GPv2Order.Data memory order = defaultOrder();
+        order.kind = GPv2Order.KIND_SELL;
+        computeSettlementForOrder(order);
+
+        bytes memory orderUid = order.computeOrderUid(domainSeparator, trader.addr);
+        assertEq(settlement.filledAmount(orderUid), order.sellAmount);
+    }
+
+    function test_should_fill_full_buy_amount_for_fill_or_kill_buy_orders() public {
+        GPv2Order.Data memory order = defaultOrder();
+        order.kind = GPv2Order.KIND_BUY;
+        computeSettlementForOrder(order);
+
+        bytes memory orderUid = order.computeOrderUid(domainSeparator, trader.addr);
+        assertEq(settlement.filledAmount(orderUid), order.buyAmount);
+    }
+
+    function test_should_fill_executed_amount_for_partially_filled_sell_orders() public {
+        GPv2Order.Data memory order = defaultOrder();
+        order.kind = GPv2Order.KIND_SELL;
+        order.partiallyFillable = true;
+        executedAmount = order.sellAmount / 3;
+        computeSettlementForOrder(order);
```

Comment:
More a general comment than a comment on this PR, since I didn't spot it on previous reviews.
I don't like using global variables to store information that is acted upon by other functions, like `executedAmount` and `computeSettlementForOrder`. This is because it's hard to determine where `executedAmount` comes from and whether other function are also impacted by this change.

I believe the reason that this happened is that JS allows to list many parameters, most of which optional, while doing the same in Solidity would be very cumbersome. Still, here I'd have probably preferred two definitions for `computeSettlementForOrder`, one with just the order as the input and the other with the executed amount as well.

But at this point I don't think a refactor is worth the time.